### PR TITLE
[MRG] participants: remove group, write json for sex+age

### LINKED ISSUE #133: Write a participants.json file
The BIDS-validator raises a warning for each column in a TSV file that has a non-custom label. Custom labels are all those labels that are explicitly defined in the specification, such as `duration` and `onset` in the events.tsv file.

We are getting the following warning:

```
	1: [WARN] Tabular file contains custom columns not described in a data dictionary (code: 82 - CUSTOM_COLUMN_WITHOUT_DESCRIPTION)
		./participants.tsv
			Evidence: Columns: age, sex, group not defined, please define in: /participants.json

```

So whenever we write age, sex, or group to our participants TSV file, we should also create an accompanying JSON file that elaborates on these column labels.

### PR DESCRIPTION
**Pull request by sappelhoff**
closes #133 

I removed the `group` column in the participants.tsv --> we are not extracting anything sensible at this moment and thus, we cannot write a proper participants.json for it (specifically, we cannot define the "levels" of the group factor and what they mean).

Else, this should be straight forward

### CODE REVIEW

**Comment by jasmainak** [comment]
Fine by me. Perhaps @monkeyman192 should take a look ...

**Comment by monkeyman192** [comment]
LGTM 👍 
I think the reason I had 'U' instead of 'n/a' was for 'unspecified'. I thought I had seen `M`, `F` and `U` in the BIDS spec, but now that I look I can't find it so maybe I just imagined it...

One thought I did have, should we maybe think about splitting up some of these functions into new files? `mne_bids.py` is starting to get quite big... Maybe we could split `tsv` and `json` writing functions into separate files or something?
This obviously wouldn't be within the scope of this PR, but it made me think of this...

**Comment by jasmainak** [comment]
@sappelhoff CIs are not happy ...

File: mne_bids/mne_bids.py [hunk 1]
```diff
@@ -658,8 +686,9 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
     manufacturer = MANUFACTURERS.get(ext, 'n/a')
 
     # save all meta data
-    _participants_tsv(raw, subject_id, "n/a", participants_fname, overwrite,
+    _participants_tsv(raw, subject_id, participants_tsv_fname, overwrite,
                       verbose)
+    _participants_json(participants_json_fname, overwrite, verbose)
```
Comment (monkeyman192) [inline mne_bids/mne_bids.py]: `overwrite = True` should fix at least one of the issues the CI's have...

**Comment by sappelhoff** [comment]
@monkeyman192 regarding the `overwrite=True`, I think that this should rather be passed here:

https://github.com/mne-tools/mne-bids/blob/828778d5ec876f21c99d50157ab1057e33f46c69/mne_bids/tests/test_mne_bids.py#L65-L76

I mean, line 76 in my posted snippet should have `overwrite True` (the `test_fif` function)


---

@jasmainak the errors are due to the following: we want to pin bids-validator ... however, we already depend on changes that were made in master ... so we need to wait for the release of 1.0.6, see: https://github.com/bids-standard/bids-validator/releases

alternatively, I can remove the pinning of the validator from this PR and we continue to work with the master of bids-validator until 1.0.6 is released - and then we pin it immediately.

**Comment by jasmainak** [comment]
okay then we merge #132 first. We should try to maximize the time CIs are happy ...

**Comment by monkeyman192** [comment]
@sappelhoff but the `participants.json` should *always* be overwritten, or never overwritten but not raise an error (which would require changing the json writing function, so this isn't viable).
The error isn't occuring just because of that test, that error will happen any time you run the script and add a new participant.

**Comment by jasmainak** [comment]
Thanks @sappelhoff ! Things are moving :-)

**Comment by sappelhoff** [comment]
yep! MNE 0.17 is out, and bids-validator 1.0.6 probably tomorrow-ish :-)

**Comment by jasmainak** [comment]
amazing !

File: mne_bids/mne_bids.py [hunk 2]
```diff
@@ -297,6 +295,34 @@ def _participants_tsv(raw, subject_id, group, fname, overwrite=False,
...
+        Whether to overwrite the existing data in the file.
+        If there is already data for the given `fname` and overwrite is False,
+        an error will be raised.
+    verbose : bool
+        Set verbose output to true or false.
+
+    """
+    cols = OrderedDict()
+    cols['participant_id'] = {'Description': 'Unique participant identifier'}
+    cols['age'] = {'Description': 'Unique participant identifier',
```
Comment (monkeyman192) [inline mne_bids/mne_bids.py]: So I clearly did a bad job of reviewing this PR... But this line is wrong. You have the same description for and and ID.

Comment (jasmainak) [reply mne_bids/mne_bids.py]: indeed! Fix is welcome :)

Comment (sappelhoff) [reply mne_bids/mne_bids.py]: :grimacing: thanks for catching @monkeyman192 !

**Pull request merged**